invisible_cities/io/mcinfo_io_test.py

```diff
@@ -0,0 +1,146 @@
+import os
+import numpy  as np
+import tables as tb
+
+from .  mcinfo_io import load_mchits
+from .  mcinfo_io import load_mcparticles
+from .  mcinfo_io import load_mcsensor_response
+from .  mcinfo_io import mc_info_writer
+from .  mcinfo_io import read_mcinfo_evt
+
+from .. reco.tbl_functions import get_mc_info
+
+from pytest import mark
+parametrize = mark.parametrize
+
+
+@mark.serial
+@parametrize('skipped_evt, out_filename',
+                  ((0, 'test_mcinfo_skip_evt0.h5'), (1, 'test_mcinfo_skip_evt1.h5')))
+def test_non_consecutive_events(config_tmpdir, ICDIR, krypton_MCRD_file, skipped_evt, out_filename):
+    filein  = os.path.join(ICDIR, 'database/test_data/', krypton_MCRD_file)
+    fileout = os.path.join(config_tmpdir, out_filename)
+
+    with tb.open_file(filein) as h5in:
+        with tb.open_file(fileout, 'w') as h5out:
+
+            mc_writer = mc_info_writer(h5out)
+            mc_extents = h5in.root.MC.extents
+            events_in = np.unique(h5in.root.MC.extents[:]['evt_number'])
+
+            mc_info = get_mc_info(h5in)
+
+            #Skip the desired event (there are only 3 in the file)
+            events_to_copy = [evt for evt in events_in if evt != skipped_evt]
```

Comment:
`events_to_copy = list(filter(lambda x: x != skipped_evt, events_in))`

Reply:
That was my other option :-), but I thought that this one was clearer.

Reply:
I think both of them are clear. Use what you like the most. `filter` tells you exactly what are you trying to do without actually reading the code, but in such a simple thing it doesn't make much of a difference.